File: spring5-demo/src/main/java/com/example/demo/classic/services/UserRepository.java
package com.example.demo.classic.services;

import com.example.demo.model.User;
import org.springframework.stereotype.Component;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.util.Arrays;
import java.util.List;
import java.util.Optional;

@Component
public class UserRepository {
    private static List<User> users = Arrays.asList(new User("Joe", 28), new User("Marc", 34));

    public Mono<User> findOneUser(String name) {
        Optional<User> userFound = users.stream().filter(user -> user.getName().equals(name)).findFirst();
        return Mono.justOrEmpty(userFound);
    }

    public Flux<User> findAllUsers() {
        User[] array = (User[]) users.toArray();
        return Flux.fromArray(array);
    }
}

File: spring5-demo/src/main/java/com/example/demo/functional/services/UserRepository.kt
package com.example.demo.functional.services

import com.example.demo.model.User
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono
import reactor.core.publisher.toFlux
import reactor.core.publisher.toMono

class UserRepository {
    private val users = arrayOf(User("Joe", 28), User("Marc", 34))

    fun findOneUser(name: String): Mono<User> = users.find { it.name == name }?.toMono() ?: Mono.empty()
    fun findAllUsers(): Flux<User> = users.toFlux()
}